lxd/patches.go

```diff
@@ -476,7 +486,15 @@ func upgradeFromStorageTypeBtrfs(name string, d *Daemon, defaultPoolName string,
 		if shared.PathExists(oldImageMntPoint) {
 			err := os.Rename(oldImageMntPoint, newImageMntPoint)
 			if err != nil {
-				return err
+				err = btrfsSnapshot(oldImageMntPoint, newImageMntPoint, false)
+				if err != nil {
+					return err
+				}
+				err = btrfsSubVolumesDelete(oldImageMntPoint)
+				if err != nil {
+					btrfsSubVolumesDelete(newImageMntPoint)
+					return err
+				}
```

Comment:
Same here. btrfs snapshot can't be cross device so I'm not sure when rename would fail but snapshot wouldn't.